scripts/generateAssetData/utils/index.ts

```diff
@@ -9,50 +18,74 @@ export const filterOutBlacklistedAssets = (unfilteredAssetData: Asset[]) =>
     return !(blacklist.includes(asset.assetId) || asset.name.toLowerCase().includes('wormhole'))
   })
 
-export const createThrottle = ({
-  capacity,
-  costPerReq,
-  drainPerInterval,
-  intervalMs,
-}: {
-  capacity: number
-  costPerReq: number
-  drainPerInterval: number
-  intervalMs: number
-}) => {
-  let currentLevel = 0
-  let pendingResolves: ((value?: unknown) => void)[] = []
-
-  const drain = () => {
-    const drainAmount = Math.min(currentLevel, drainPerInterval)
-    currentLevel -= drainAmount
-
-    // Resolve pending promises if there's enough capacity
-    while (pendingResolves.length > 0 && currentLevel + costPerReq <= capacity) {
-      const resolve = pendingResolves.shift()
-      if (resolve) {
-        currentLevel += costPerReq
-        resolve()
-      }
-    }
-  }
+const getAssetIdsSortedByMarketCap = async (): Promise<AssetId[]> => {
+  let page = 1
+
+  const results2d: CoinGeckoMarketCap[][] = []
+
+  while (true) {
+    const coingeckoAssets = await getCoingeckoMarketsRaw('market_cap_desc', page, 100)
```

Comment:
blocking: this will ensure we have more per pages, resulting in less pages fetched
```suggestion
    const coingeckoAssets = await getCoingeckoMarketsRaw('market_cap_desc', page, 250)
```

<img width="584" alt="image" src="https://github.com/user-attachments/assets/b338f2d4-1afa-41a4-926d-a1c1043431eb" />

https://docs.coingecko.com/reference/coins-markets

Reply:
Let er rip f83b54f229f582a8e120ef6666483d5d6891de11